app/javascript/app/document-capture/components/file-input.jsx

```diff
@@ -33,15 +89,19 @@ export function toDataURL(file) {
   });
 }
 
-function FileInput({ label, hint, bannerText, accept, value, onChange, className }) {
+function FileInput({ label, hint, bannerText, accept, value, errors, onChange, className }) {
   const { t, formatHTML } = useI18n();
   const ifStillMounted = useIfStillMounted();
   const instanceId = useInstanceId();
   const { isMobile } = useContext(DeviceContext);
   const [isDraggingOver, setIsDraggingOver] = useState(false);
+  const [ownErrors, setOwnErrors] = useState(/** @type {string[]} */ ([]));
+  useMemo(() => setOwnErrors([]), [value]);
```

Comment:
Not entirely obvious, but `useMemo` is an ideal fit here, since unlike `useEffect` it is called _immediately_ at render-time, not _after_ the render is applied to the DOM. The intent is to clear whatever errors might have existed any time the value is changed, which is why `[value]` is defined as the dependency.